src/lambda/local/debugConfiguration.ts

```diff
@@ -92,3 +109,73 @@ export function assertTargetKind(config: SamLaunchRequestArgs, expectedTarget: '
         throw Error(`SAM debug: invalid config: ${config}`)
     }
 }
+
+export function getCodeRoot(
```

Comment:
Are there tests for this function? I don't see them in this PR, but it looks like this function moved from one file to another, so maybe they do, and need to be moved as well.

I'm interested in tests for the following cases:
* target is template and `templateResource?.Properties?.CodeUri` contains an absolute path or a relative path

Reply:
created a tracking issue. This gets full coverage indirectly because it is a core part of the "config" phase tested in this PR. I think more direct coverage should not blocking merging this PR, because this PR is blocking 2 other PRs.